templates/flavors/rke2/kustomization.yaml

```diff
@@ -14,6 +15,17 @@ patches:
       - op: replace
         path: /spec/controlPlaneRef/kind
         value: RKE2ControlPlane
+  - target:
+      group: cluster.x-k8s.io
+      version: v1beta1
+      kind: Cluster
+    patch: |-
+      apiVersion: cluster.x-k8s.io/v1beta1
+      kind: Cluster
+      metadata:
+        name: ${CLUSTER_NAME}
+        labels:
+            cni: cilium
```

Comment:
Would it be better to use a `helm.cattle.io/v1/HelmChart` resource on the workload cluster to manage it's own Cilium Helm chart instead of [CAAPH](https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm) here?

I see we're using already using this^ pattern for the CCM and for Cilium on the K3s flavor.

Reply:
I am not sure about which one is better. We currently have mixed implementations and would like to pick one or the other.